### PR TITLE
FUSETOOLS-3013 - added better validation for the Ignite Version field

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [x] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

File: syndesis/plugins/org.fusesource.ide.syndesis.extension.core/src/org/fusesource/ide/syndesis/extensions/core/util/SyndesisExtensionsUtil.java [hunk 1]
```diff
@@ -31,11 +31,15 @@ public static boolean isValidSyndesisExtensionVersion(String version) {
 		if (parts.length<2 || version.trim().endsWith(".") || version.trim().startsWith(".")) {
 			valid = false;
 		} else {
+			int foundNumbers = 0;
 			for (String part : parts) {
 				try {
 					Integer.parseInt(part);
+					foundNumbers++;
```
Comment (apupier) [inline syndesis/plugins/org.fusesource.ide.syndesis.extension.core/src/org/fusesource/ide/syndesis/extensions/core/util/SyndesisExtensionsUtil.java]: I don't get from the code why this foundNumbers has been introduced. I think adding a test might help to understand it and avoid regression

**Comment by lhein** [comment]
I improved the test logic and added several new test params to cover most usecases / mistakes.

File: syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java [hunk 2]
```diff
@@ -30,14 +30,19 @@
 		return new String[] {
 				"1",
 				"1-SNAPSHOT",
-				"1.1.SNAPSHOT",
```
Comment (apupier) [inline syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java]: why this use case has been removed?

File: syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java [hunk 3]
```diff
@@ -30,14 +30,19 @@
...
-				"1.1-SNAPSHOT",
-				"1.1.0-SNAPSHOT",
```
Comment (apupier) [inline syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java]: why this was invalid before and is valid now?

File: syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java [hunk 4]
```diff
@@ -30,14 +30,19 @@
 		return new String[] {
 				"1",
 				"1-SNAPSHOT",
-				"1.1.SNAPSHOT",
-				"1.1-SNAPSHOT",
```
Comment (apupier) [inline syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java]: why this was invalid before and is valid now?

File: syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java [hunk 5]
```diff
@@ -30,14 +30,19 @@
...
-				"1.1.0-SNAPSHOT",
-				"1.1.v20180108",
```
Comment (apupier) [inline syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java]: why this use case has been removed?

File: syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java (hunk 3 above)
Comment (lhein) [reply syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java]: We need valid maven versions here. As this value is a valid maven version it has to be removed from the InvalidVersionsTest.

File: syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java (hunk 2 above)
Comment (lhein) [reply syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java]: added it back

File: syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java (hunk 4 above)
Comment (lhein) [reply syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java]: We need valid maven versions here. As this value is a valid maven version it has to be removed from the InvalidVersionsTest.

File: syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java (hunk 5 above)
Comment (lhein) [reply syndesis/tests/org.fusesource.ide.syndesis.extension.core.tests/src/test/java/org/fusesource/ide/syndesis/extensions/core/model/SyndesisExtensionUtilInvalidVersionTest.java]: We need valid maven versions here. As this value is a valid maven version it has to be removed from the InvalidVersionsTest.

**Comment by lhein** [comment]
I improved the error message.

For t he shared validation logic I think it is not really needed atm. We just need to check for a valid maven version and that is what my logic does.

**Pull request merged**